web/frontend/src/api/navigation.ts

```diff
@@ -33,44 +38,55 @@ export const setWaypoint = (client: Client, lat: number, lng: number, name: stri
   rcLogConditionally(request);
 
   return new Promise((resolve, reject) => {
-    client.navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) =>
+    get(client).navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) =>
       (error ? reject(error) : resolve(response)));
   });
 };
 
-export const getWaypoints = (client: Client, name: string) => {
+const transformWaypoints = (list: navigationApi.Waypoint[]) => {
+  return list.map((item) => {
+    const location = item.getLocation();
+    return {
+      id: item.getId(),
+      lng: location?.getLongitude() ?? 0,
+      lat: location?.getLatitude() ?? 0,
+    };
+  });
+};
+
+export const getWaypoints = (name: string) => {
   const req = new navigationApi.GetWaypointsRequest();
   req.setName(name);
 
   rcLogConditionally(req);
 
-  return new Promise<navigationApi.Waypoint[]>((resolve, reject) => {
-    client.navigationService.getWaypoints(req, new grpc.Metadata(), (error, response) =>
-      (error ? reject(error) : resolve(response?.getWaypointsList() ?? [])));
+  return new Promise<Waypoint[]>((resolve, reject) => {
+    get(client).navigationService.getWaypoints(req, new grpc.Metadata(), (error, response) =>
+      (error ? reject(error) : resolve(transformWaypoints(response?.getWaypointsList() ?? []))));
```

Comment:
[q] why does the waypoint list need to be transformed? has the function signature changes to output something new which needs to be processed?

Reply:
The data that this GRPC request returns is needlessly complicated / verbose / the data structure sucks. This is just me containing the chaos within this `api` folder before it hits `components` - a strategy we've been employing in app for a while now.

Renamed this `formatWaypoints` to make it less confusing!